File: src/2025-02-23-deepclaude-try/javascript/exercises/practice/phone-number/phone-number.js
//
// This is only a SKELETON file for the 'Phone Number' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const clean = (number) => {
  // Check for letters
  if (/[a-zA-Z]/.test(number)) {
    throw new Error('Letters not permitted');
  }

  // Check for invalid punctuations (only allow +, -, ., space, parentheses)
  if (/[^\d+()\- .]/.test(number)) {
    throw new Error('Punctuations not permitted');
  }

  // Extract digits and check length
  let digits = number.replace(/\D/g, '');
  const numDigits = digits.length;

  if (numDigits < 10) {
    throw new Error('Incorrect number of digits');
  } else if (numDigits > 11) {
    throw new Error('More than 11 digits');
  }

  // Check country code
  if (numDigits === 11) {
    if (digits[0] !== '1') {
      throw new Error('11 digits must start with 1');
    }
    digits = digits.substring(1);
  }

  // Validate area code
  const areaCode = digits.substring(0, 3);
  if (areaCode[0] === '0') {
    throw new Error('Area code cannot start with zero');
  } else if (areaCode[0] === '1') {
    throw new Error('Area code cannot start with one');
  }

  // Validate exchange code
  const exchangeCode = digits.substring(3, 6);
  if (exchangeCode[0] === '0') {
    throw new Error('Exchange code cannot start with zero');
  } else if (exchangeCode[0] === '1') {
    throw new Error('Exchange code cannot start with one');
  }

  return digits;
};

File: src/2025-02-23-deepclaude-try/java/exercises/practice/phone-number/src/main/java/PhoneNumber.java
class PhoneNumber {
    private String number;

    PhoneNumber(String numberString) {
        if (numberString == null) {
            throw new IllegalArgumentException("Phone number cannot be null");
        }
        
        // Clean the input - remove all non-digits
        String cleaned = numberString.replaceAll("[^0-9]", "");
        
        // Handle 11-digit numbers that start with 1
        if (cleaned.length() == 11 && cleaned.startsWith("1")) {
            cleaned = cleaned.substring(1);
        }
        
        // Validate length
        if (cleaned.length() != 10) {
            throw new IllegalArgumentException("Invalid phone number length");
        }
        
        // Validate area code and exchange code (first and fourth digits must be 2-9)
        if (cleaned.charAt(0) < '2' || cleaned.charAt(0) > '9' ||
            cleaned.charAt(3) < '2' || cleaned.charAt(3) > '9') {
            throw new IllegalArgumentException("Invalid area code or exchange code");
        }
        
        this.number = cleaned;
    }

    String getNumber() {
        return number;
    }

}
